airbyte-integrations/connectors/source-bigcommerce/source_bigcommerce/source.py

```diff
@@ -167,19 +167,6 @@ def path(self, **kwargs) -> str:
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
         return response.json() if len(response.content) > 0 else []
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        max_date = ""
-        if not current_stream_state.get(self.cursor_field):
-            max_date = latest_record.get(self.cursor_field, "")
-        elif latest_record.get(self.cursor_field) and current_stream_state.get(self.cursor_field):
-            latest_state_date_time = parsedate_tz(latest_record.get(self.cursor_field))
-            current_state_date_time = parsedate_tz(current_stream_state.get(self.cursor_field))
-            max_date = current_stream_state.get(self.cursor_field)
-            if current_state_date_time < latest_state_date_time:
-                max_date = latest_record.get(self.cursor_field)
-
-        return {self.cursor_field: max_date}
-
```

Comment:
this is on purpose ?

Reply:
Yes, because the [transformation](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-bigcommerce/source_bigcommerce/source.py#L40) introduced in #9516  takes a datetime in `RFC-2822` and converts it to `ISO 8601`.   
The purpose of this function, IIUC, is to compare dates in `RFC-2822`.

`parsedate_tz` is a function from `email.utils`, which uses `email.utils.parsedate` under the hood:
> email.utils.parsedate(date)
> Attempts to parse a date according to the rules in RFC 2822....

Does the deletion make sense to you?